### PR TITLE
Logging in workers

### PR DESCRIPTION
**Pull request by pawbu**
- based on https://docs.python.org/3/howto/logging-cookbook.html#logging-to-a-single-file-from-multiple-processes
- logging from other processes is done in the main process, in a separate thread
- json-logging dependency used to produce logs in JSON format
- format of the logs follows the one in other services we have minus the web stuff
- used Wiremock to easily stub out collaborating services
- added more logs (timing of the workers ++)
- a ContextFilter adds a prefix with process name for each worker so that logs from different workers are distinguishable and at the same time it works in our Kibana

### CODE REVIEW

File: job_executor/config/log.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        self.process_name = process_name
+
+    def filter(self, record):
+        # prefix record.msg instead of adding a new field
+        # to be compliant with Kibana
+        record.msg = self.process_name + ': ' + record.msg
+        return True
+
+
+def configure_worker_logger(queue: Queue):
```
Comment (DanielElisenberg) [inline job_executor/config/log.py:32]: Is there a way we could get the jobId for the running job in the log here too? 🤔
(Does not need to be part of this PR, but maybe a task for the future)

File: job_executor/worker/build_metadata_worker.py [hunk 2]
```diff
@@ -43,3 +53,9 @@ def run_worker(job_id: str, dataset_name: str):
         job_service.update_job_status(
             job_id, 'failed', log='Unexpected exception when building dataset'
         )
+    finally:
+        delta = perf_counter() - start
```
Comment (DanielElisenberg) [inline job_executor/worker/build_metadata_worker.py:57]: :100:

File: job_executor/worker/steps/dataset_pseudonymizer.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def _pseudonymize_identifier_only(
             target_file.write(
                 ';'.join([
                     line_number,
-                    identifier_to_pseudonym[unit_id],
+                    str(identifier_to_pseudonym[unit_id]),
```
Comment (DanielElisenberg) [inline job_executor/worker/steps/dataset_pseudonymizer.py:48]: :100:

**Review by DanielElisenberg — APPROVED** [review]
Looks really good 🚀  Would love to talk about this in depth when I've read the link and there's time

File: job_executor/config/log.py (hunk 1 above)
Comment (pawbu) [reply job_executor/config/log.py:32]: Sure, I can add it as a prefix to message instead of process name as I don't think we need both.

**Pull request merged**